tests/test_managers.py

```diff
@@ -1,62 +1,61 @@
+import sys
+import os
 import asyncio
-import unittest
-from unittest.mock import AsyncMock, patch
+from unittest.mock import AsyncMock
 
-from fastapi.testclient import TestClient
-from managers import ConnectionManager, Message
-from main import app
+import pytest
+from tests.conftest import client
 
 
-class WebSocketTests(unittest.TestCase):
-    def setUp(self):
-        self.client = TestClient(app)
-        self.manager = ConnectionManager()
-        self.websocket = AsyncMock()
+sys.path.append(os.path.dirname(os.path.dirname(__file__)))
 
-    def test_websocket_connect(self):
-        asyncio.run(self.manager.connect(self.websocket))
-        self.assertIn(self.websocket, self.manager.active_connections)
+from fastcrawler_ui.repository.ws import ConnectionRepository, Message
 
-    def test_connection_manager_broadcast(self):
-        websocket_mock1 = AsyncMock()
-        websocket_mock2 = AsyncMock()
 
-        self.manager.active_connections.add(websocket_mock1)
-        self.manager.active_connections.add(websocket_mock2)
 
-        message_dict = {"content": "Hello", "sender": "Sadegh"}
-        message = Message(**message_dict)
 
-        asyncio.run(self.manager.broadcast(message))
+def test_websocket_connect(manager: ConnectionRepository, websocket):
+    asyncio.run(manager.connect(websocket))
+    assert websocket in manager.active_connections
 
-        websocket_mock1.send_json.assert_called_once_with(message.model_dump())
-        websocket_mock2.send_json.assert_called_once_with(message.model_dump())
 
-    def test_connection_manager_send_personal_message(self):
-        websocket_mock = AsyncMock()
+def test_connection_manager_broadcast(manager: ConnectionRepository):
```

Comment:
the tests are still written with unittest. Please get rid of AsyncMock.
Instead mock the WSController.